web/src/utils/factory-management/syncState.ts

```diff
@@ -37,17 +37,18 @@ export const calculateSyncState = (factory: Factory) => {
 }
 
 export const checkProductSyncState = (factory: Factory) => {
-  // Scan the products and determine if they are in sync with the syncState.
-  // If there are no products, mark the factory out of sync, if already in sync.
-  if (!factory.products.length) {
-    if (factory.inSync) {
-      factory.inSync = false
-    }
+  // If factory has no products or power producers, mark as out of sync.
+  if (!factory.products.length && !factory.powerProducers.length) {
+    factory.inSync = false
   }
 
-  // If the number of products is different from the syncState, mark the factory as out of sync.
+  // If the number of products differs from syncState, mark as out of sync.
+  // Exception: fuel-only factories (no products, no syncState, but have power producers) remain in sync.
   if (factory.products.length !== Object.keys(factory.syncState).length) {
-    factory.inSync = false
+    const isFuelOnlyFactory = factory.products.length === 0 && Object.keys(factory.syncState).length === 0 && factory.powerProducers.length > 0
```

Comment:
```suggestion
    const isFuelOnlyFactory = factory.products.length === 0 && factory.powerProducers.length > 0
```

You shouldn't need to care about sync state at this point.

Reply:
@copilot